src/osd/ReplicatedBackend.cc

```diff
@@ -1595,21 +1612,66 @@ void ReplicatedBackend::submit_push_data(
   if (first) {
     t->remove(coll, ghobject_t(target_oid));
     t->touch(coll, ghobject_t(target_oid));
+ if (!complete) {
+      bufferlist bv = attrs.at(OI_ATTR);
+      object_info_t oi(bv);
+      t->set_alloc_hint(coll, ghobject_t(target_oid),
+                        oi.expected_object_size,
+                        oi.expected_write_size,
+                        oi.alloc_hint_flags);
+    } else {
+      //remove xattr and update later if overwrite on original object
+      t->rmattrs(coll, ghobject_t(target_oid));
+      //if need update omap, clear the previous content first
+      if (clear_omap)
+        t->omap_clear(coll, ghobject_t(target_oid));
```

Comment:
removing attrs and omap only needs to be done for the first push op, not every one